syntax="proto3";
package pb;
//无关选项，用于客户端c#指定命名空间
option csharp_namespace="Pb";

//LOGON_SYNCPID 和 LOGOFF_SYNCPID
message SyncPid{
    int32 Pid=1;
    string Username=2;
}

message Player{
    int32 Pid=1;
    Position P=2;
    string Username=3;
}

// SUR_PLAYER
message SyncPlayers{
    /*嵌套多个子消息类型Player的消息  repeated 表示可重复，类似定义数组*/ 
    repeated Player ps=1;
}

//NEW_POSTION
message Position{
    float X=1;
    float Y=2;
    float Z=3;
    float V=4;
    int32 BloodValue=5;
}

message MovePackege{
    Position P=1;
    int32 ActionData=2;
}

//BROADCAST
message BroadCast{
    int32 Pid=1;
    int32 Tp=2;
    /*根据Tp不同，BroadCast消息会包含：
      1 聊天内容（Content）
      2 初始位置(P)
      4 新位置P*/
    oneof Data {
        string Content=3;
        Position P=4;
        /*ActionData暂时预留*/
        int32 ActionData=5;
    }
    string Username=6;
}


//GAME_MSG_TALK_CONTENT
message Talk{
    string Content=1;
}

